tools/integration-test/src/tests/fee/non_fee_channel.rs

```diff
@@ -0,0 +1,100 @@
+use ibc_test_framework::prelude::*;
+use ibc_test_framework::util::random::random_u128_range;
+
+#[test]
+fn test_non_fee_channel() -> Result<(), Error> {
+    run_binary_channel_test(&NonFeeChannelTest)
+}
+
+struct NonFeeChannelTest;
+
+impl TestOverrides for NonFeeChannelTest {
+    fn modify_relayer_config(&self, config: &mut Config) {
+        config.mode.packets.auto_register_counterparty_payee = true;
+    }
+}
+
+impl BinaryChannelTest for NonFeeChannelTest {
+    fn run<ChainA: ChainHandle, ChainB: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        _relayer: RelayerDriver,
+        chains: ConnectedChains<ChainA, ChainB>,
+        channel: ConnectedChannel<ChainA, ChainB>,
+    ) -> Result<(), Error> {
+        let chain_driver_a = chains.node_a.chain_driver();
+        let chain_driver_b = chains.node_b.chain_driver();
+
+        let denom_a = chains.node_a.denom();
+
+        let port_a = channel.port_a.as_ref();
+        let port_b = channel.port_b.as_ref();
+        let channel_id_a = channel.channel_id_a.as_ref();
+        let channel_id_b = channel.channel_id_b.as_ref();
+
+        let wallets_a = chains.node_a.wallets();
+        let wallets_b = chains.node_b.wallets();
+
+        let user_a = wallets_a.user1();
+        let user_b = wallets_b.user1();
+
+        let relayer_a = wallets_a.relayer();
+        let relayer_b = wallets_b.relayer();
+
+        let balance_a1 = chain_driver_a.query_balance(&user_a.address(), &denom_a)?;
+
+        let send_amount = random_u128_range(1000, 2000);
+
+        {
+            let res = chain_driver_b.register_counterparty_payee(
+                &relayer_b,
+                &relayer_a.address(),
+                &channel_id_b,
+                &port_b,
+            );
+
+            assert!(res.is_err());
```

Comment:
I'm a bit unclear on how this test works. Is `register_counterparty_payee` returning an error because `auto_register_counterparty_payee` is `true`, or is it failing for a different reason?

Reply:
My understanding is that it fails because the channel compatible with ICS29 fees, the `channel_version()` override to set the version to `ics20_with_fee()` is left out, https://github.com/informalsystems/hermes/blob/df077e6e0c634e13cd9bd19d10e648362a09976c/tools/integration-test/src/tests/fee/non_fee_channel.rs#L11-L15 and will thus default to the version without fee, https://github.com/informalsystems/hermes/blob/df077e6e0c634e13cd9bd19d10e648362a09976c/tools/test-framework/src/framework/overrides.rs#L146
So there will be an Unknown message error for the `MsgRegisterCounterpartyPayee` when trying to register the counterparty payee.